pkg/operator/ceph/cluster/controller.go

```diff
@@ -341,17 +344,46 @@ func (c *ClusterController) onK8sNodeUpdate(oldObj, newObj interface{}) {
 		logger.Warningf("Expected Node but handler received %#v", oldObj)
 		return
 	}
+	newNodeSchedulable := k8sutil.GetNodeSchedulable(*newNode)
+	oldNodeSchedulable := k8sutil.GetNodeSchedulable(*oldNode)
+	clusters := c.getTenantClusters(newNode)
 
-	if k8sutil.GetNodeSchedulable(*newNode) == false {
-		logger.Debugf("Skipping cluster update. Updated node %s is unschedulable", newNode.Labels[apis.LabelHostname])
-		return
+	for _, cluster := range clusters {
+		logger.Debugf("NodeUpdate: oldNodeSchedulable: %v, newNodeSchedulable: %v", oldNodeSchedulable, newNodeSchedulable)
+		if cluster.Info.IsInitialized() {
+			nodes, err := GetAllStorageNodes(cluster.context, cluster.Namespace)
+			if err != nil {
+				logger.Errorf("Error getting all storage nodes for cluster: %v", err)
+			}
+			allSchedulable := true
+			for _, node := range nodes {
+				if !k8sutil.GetNodeSchedulable(node) {
+					allSchedulable = false
+					break
+				}
+			}
+			if allSchedulable {
+				logger.Infof("Unsetting noout because a storage node is in maintenance")
+				client.UnsetNoOut(c.context, cluster.Info.Name)
+			} else {
+				logger.Infof("Setting noout because no storage nodes are in maintenance")
```

Comment:
Should this message be?
"Setting noout because at least one node is in maintenance"